test/lib/eventFactory.test.js

```diff
@@ -142,6 +157,184 @@ describe('Event Factory', () => {
             datastore.save.resolves({ id: 'xzy1234' });
         });
 
+        describe('with new workflow', () => {
+            beforeEach(() => {
+                jobsMock = [{
+                    id: 1,
+                    pipelineId: 8765,
+                    name: 'component',
+                    permutations: {
+                        requires: ['~commit', '~pr']
+                    },
+                    state: 'ENABLED'
+                }, {
+                    id: 2,
+                    pipelineId: 8765,
+                    name: 'disabledjob',
+                    permutations: {
+                        requires: ['component']
+                    },
+                    state: 'DISABLED'
+                }, {
+                    id: 3,
+                    pipelineId: 8765,
+                    name: 'integration',
+                    permutations: {
+                        requires: ['test', '~pr']
+                    },
+                    state: 'ENABLED'
+                }, {
+                    id: 4,
+                    pipelineId: 8765,
+                    name: 'deploy',
+                    permutations: {
+                        requires: ['integration']
+                    },
+                    state: 'ENABLED'
+                }];
+
+                pipelineMock.jobs = Promise.resolve(jobsMock);
+                buildFactoryMock.create.resolves(null);
+            });
+
+            it('should start pr builds if using new workflow', () => {
+                jobsMock = [{
+                    id: 5,
+                    pipelineId: 8765,
+                    name: 'PR-1-component',
+                    permutations: {
+                        requires: ['~pr']
+                    },
+                    state: 'ENABLED'
+                }, {
+                    id: 6,
+                    pipelineId: 8765,
+                    name: 'PR-1-integration',
+                    permutations: {
+                        requires: ['~pr']
+                    },
+                    state: 'DISABLED'
+                }];
+
+                pipelineMock.jobs = Promise.resolve(jobsMock);
+
+                const prComponent = {
+                    id: 5,
+                    name: 'PR-1-component'
+                };
+
+                jobFactoryMock.create.onCall(0).resolves(prComponent);
+                config.startFrom = '~pr';
+                config.prRef = 'branch';
+                config.prNum = '1';
+
+                return factory.create(config).then((model) => {
+                    assert.instanceOf(model, Event);
+                    assert.calledOnce(buildFactoryMock.create);
```

Comment:
This test should have failed, if comment above is addressed. Also see potential fix I've given in above.